### DISCUSSION TITLE
Question about KMK Hold-Tap implementation of Super-Key under Windows and/or Linux

### DISCUSSION THREAD

**Discussion by psycho-pate** [manna-harbour/miryoku#218] [Q&A] [answered]
Hello everyone,

I have been working on building a PicoSplit keyboard using the framework provided by https://github.com/akaenner/picosplit. It should be similar to the Miryoku KMK if i understood it correctly and designed for macOS. However, I am experiencing an issue with the behavior of the modifier keys (Super and Alt) when using the keyboard on Windows and Linux.

The problem I am encountering is that whenever I try to input the letter 'A', it triggers the Start Menu, and when I try to input the letter 'S', it opens a program menu. It seems that the open-source driver for PicoSplit immediately triggers the modifier keys instead of waiting to see if I intend to activate a shortcut.

Since i'm not a programmer, i have reached out to the developer of PicoSplit, but since they primarily use macOS, they are not familiar with the proper implementation for Windows and Linux. They have expressed their willingness to fix this issue if provided with the correct implementation details.

Therefore, I would like to kindly request assistance from the Miryoku KMK community. If anyone has experience with configuring modifier keys behavior on Windows and Linux, I would greatly appreciate your guidance and suggestions. It would be immensely helpful if you could share insights on how this behavior is correctly implemented in the Miryoku KMK framework.

Thank you in advance for your support and expertise!

**Comment by manna-harbour** [comment, marked as answer]
Making mod-tap usable on alphas requires sufficient configuration support in the firmware as well as an appropriate configuration in the keymap.

See the hold-tap documentation in other firmware for details:

- https://github.com/qmk/qmk_firmware/blob/master/docs/tap_hold.md
- https://zmk.dev/docs/behaviors/hold-tap
- https://github.com/KMKfw/kmk_firmware/blob/master/docs/en/holdtap.md

Configurations where the mod is sent immediately and then cancelled when the tap is decided aren't common due to the issue you've noticed of Windows reacting to flashing mods.

As an alternative you could make a keyboard definition for picosplit in QMK or KMK, or use single function keys only in your current firmware and implement the mod-tap based keymap in KMonad or via 𝑥MK.
